src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -673,13 +759,35 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            firstVector = Sse2.LoadVector128((ushort*)pBuffer); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                firstVector = Sse2.LoadVector128((ushort*)pBuffer); // unaligned load
+            }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                firstVector = AdvSimd.LoadVector128((ushort*)pBuffer); // unaligned load
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
 
             // The operation below forces the 0x8000 bit of each WORD to be set iff the WORD element
             // has value >= 0x0800 (non-ASCII). Then we'll treat the vector as a BYTE vector in order
             // to extract the mask. Reminder: the 0x0080 bit of each WORD should be ignored.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            if (Sse2.IsSupported)
+            {
+                currentMask = (ulong)Sse2.MoveMask(Sse2.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                currentMask = GetMaskOfMostSignificantBits(AdvSimd.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
 
             if ((currentMask & NonAsciiDataSeenMask) != 0)
```

Comment:
How is `NonAsciiDataSeenMask` check supposed to work?
Basically we want to check if any of the 16 lane contains non-ascii or not. If it was, then that bit in the mask was set. So if lane 0 has non-ascii, then bit 0 in resulting `currentMask` should be set. But from the ` & NonAsciiDataSeenMask` operation, we just check for odd positions ?

Reply:
Shouldn't it should be ` & 0xFFFF` ?